src/pudl/io_managers.py

```diff
@@ -156,8 +171,15 @@ def _setup_database(self, timeout: float = 1_000.0) -> sa.engine.Engine:
         # Create the database and schemas
         if not db_path.exists():
             db_path.touch()
-            self.md.create_all(engine)
-
+            if exclude_tables:
+                # Get sql alchemy table objects that should not be excluded
+                table_names = {table_name for table_name in self.md.tables} - set(
+                    exclude_tables
+                )
+                tables = tuple(self.md.tables[table_name] for table_name in table_names)
+                self.md.create_all(engine, tables=tables)
+            else:
+                self.md.create_all(engine)
```

Comment:
@zaneselvans would love your thoughts on this. This bit of code allows us to exclude SQL view metadata from being created as table schemas in the database. We don't want to create table schemas for views because the table and view name will conflict in the database. There are still `sa.Table` objects for the sql views in `self.md` because we need some way to make sure metadata in `pudl.metadta.resources` for a given view has been created so the types are corrected in load_input when the view is read as a dataframe.

I thought about just passing a `Package()` object to the `SQLiteIOManager` so it could filter out tables using:

```python
Package.from_resource_ids(
        excluded_etl_groups=(
            "output_views",
            ...
        )
    ).to_sql()
```

I decided against this because the `Package` class needs access to `pudl.metadata.resources` which means the `SQLiteIOManager` couldn't be used in packages outside of PUDL. #1522 describes how our metadata classes are dependent on pudl-specific constants. Using the IO Manager outside of the pudl package isn't super important right now but it would be nice to decouple our metadata from our useful metadata and io manager classes.

Reply:
This has been resolved with the changes from #2466.